### PR TITLE
Made Kubernetes version advice generic across installation guides

### PR DESCRIPTION
**Pull request by jwitz**
Closes https://github.com/astronomer/docs/issues/218

As @rbankston noted, wrote about Kubernetes compatibility in such a way that it would continually be out of date whenever we updated the platform version. I've rewritten these considerations to instead link out to the version compatibility reference guide, which should always have the current info for each release's version requirements.

### CODE REVIEW

**Review by rbankston — APPROVED** [review]
Looks Great @jwitz

File: enterprise/next/03_install/01_aws/01_install-aws-standard.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ EKS is built off of Amazon's pre-existing EC2 service, so you can manage your Ku
 
 As you follow the guide linked above, keep in mind:
 
-* Astronomer currently supports Kubernetes versions 1.14, 1.15, 1.16 and 1.17 on EKS.
+* Each version of Astronomer Enterprise has its own version requirements for Kubernetes. For more information on version compatibility, refer to the [Version Compatibility Reference](https://www.astronomer.io/docs/enterprise/stable/resources/version-compatibility-reference).
```
Comment (paolaperaza) [inline enterprise/next/03_install/01_aws/01_install-aws-standard.md:45]:
```suggestion
* The versions of Kubernetes that are compatible with the platform vary across versions of Astronomer Enterprise. For more information on version compatibility, refer to Astronomer's [Version Compatibility Reference](https://www.astronomer.io/docs/enterprise/stable/resources/version-compatibility-reference).
```

Comment (paolaperaza) [reply enterprise/next/03_install/01_aws/01_install-aws-standard.md:45]: Suggestion on copy, but nbd - feel free to take or leave!

**Review by paolaperaza — COMMENTED** [review]
@jwitz I like the idea of linking out to our version compatibility reference more prominently. I made a slight copy suggestion for the primary change in this PR, but had 1 more higher-level question:

At the top of all of these install docs, required versions of K8s are listed in the "Prerequisites" section. Did we want to remove that line, and again point people to the reference doc? It should be consistent with what we state later in the doc.

As I'm thinking about this, though, and since we don't come out with new versions of Enterprise all that often, I'd maybe consider that we:

1. We keep specific K8s version in prerequisites and make sure the guidance there is correct depending on the version of the doc
2. We keep your comment below, and expand it beyond k8s versions to more generally say something like, "Versions compatibility of each component required by Astronomer, including Kubernetes, Helm, and Apache Airflow, varies across versions of Astronomer Enterprise. For more information...."

Just my thought, but I can see it both ways.

File: enterprise/next/03_install/01_aws/01_install-aws-standard.md (hunk 1 above)
Comment (jwitz) [reply enterprise/next/03_install/01_aws/01_install-aws-standard.md:45]: Keeping most of the OG phrasing here, but like changing from "requirements" to "compatibility"

Comment (jwitz) [reply enterprise/next/03_install/01_aws/01_install-aws-standard.md]: Did a hybrid of your suggestion and the OG wording.

**Comment by jwitz** [comment]
@paolaperaza Since Kubernetes is really the only thing in the version compatibility guide that varies between versions, I decided to keep everything Kubernetes-specific and to add the Version Compatibility guide to that prereq. I think it's also smart to get that compatibility in front of people as soon as possible.

File: enterprise/next/03_install/01_aws/01_install-aws-standard.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ EKS is built off of Amazon's pre-existing EC2 service, so you can manage your Ku
 
 As you follow the guide linked above, keep in mind:
 
-* Astronomer currently supports Kubernetes versions 1.14, 1.15, 1.16 and 1.17 on EKS.
+* Each version of Astronomer Enterprise has its own Kubernetes version compatibility. For more information on version compatibility, refer to Astronomer's [Version Compatibility Reference](https://www.astronomer.io/docs/enterprise/stable/resources/version-compatibility-reference).
```
Comment (paolaperaza) [inline enterprise/next/03_install/01_aws/01_install-aws-standard.md:45]:
```suggestion
* Each version of Astronomer Enterprise is compatible with a particular set of Kubernetes versions. For more information, refer to Astronomer's [Version Compatibility Reference](https://www.astronomer.io/docs/enterprise/stable/resources/version-compatibility-reference).
```

This sentence still sounded off/passive to me - WDYT about this?

**Review by paolaperaza — APPROVED** [review]
This actually sounds good to me - I like opting for not specifying versions across the board and pointing people to our doc sooner. Made one final copy change, but feel free to merge here as soon a you review that/make a call.

**Comment by danielhoherd** [comment]
I love the links to the compatibility matrix. Great way to handle this!

**Pull request merged**